synapse/lib/lmdbslab.py

```diff
@@ -709,6 +709,8 @@ async def __anit__(self, path, **kwargs):
         kwargs.setdefault('lockmemory', False)
         kwargs.setdefault('map_async', True)
 
+        assert kwargs.get('map_async')
```

Comment:
I don't know why this is being removed; and this assertion is soft and would be dropped in a python interpreter that is running without `assert` statements.

Reply:
Dropped because it no longer matters if they set it or not

Reply:
Yeah but if someone was previously using the raw slab object in sync mode we'd be removing this out from under them. We can change this from our use perspective but it doesn't feel correct to change the slab class for existing slabs.

Reply:
We don't ever want to be blocking the ioloop on these types of write operations.